### PR TITLE
Implement CUDA kernel for Shrink operator

### PR DESCRIPTION
**Pull request by hariharans29**

### CODE REVIEW

File: onnxruntime/core/providers/cuda/nn/shrink.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+namespace cuda {
+
+template <typename T>
+class Shrink final : public CudaKernel {
+ public:
+  Shrink(const OpKernelInfo& info) : CudaKernel(info) {
+    float bias_temp;
+    // if the attribute exists, use the value
+    if (info.GetAttr<float>("bias", &bias_temp).IsOK())
+      bias_ = gsl::narrow_cast<float>(bias_temp);
```
Comment (HectorSVC) [inline onnxruntime/core/providers/cuda/nn/shrink.h]: >gsl::narrow_cast<float> [](start = 14, length = 23)

why need a cast here?

File: onnxruntime/core/providers/cuda/nn/shrink.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    float lambd_temp;
+    // if the attribute exists, use the value
+    if (info.GetAttr<float>("lambd", &lambd_temp).IsOK())
+      lambd_ = gsl::narrow_cast<float>(lambd_temp);
```
Comment (HectorSVC) [inline onnxruntime/core/providers/cuda/nn/shrink.h]: >gsl::narrow_cast<float> [](start = 15, length = 23)

same with above

File: onnxruntime/core/providers/cuda/nn/shrink.h (hunk 1 above)
Comment (hariharans29) [reply onnxruntime/core/providers/cuda/nn/shrink.h]: Fixed.

File: onnxruntime/core/providers/cuda/nn/shrink.h (hunk 2 above)
Comment (hariharans29) [reply onnxruntime/core/providers/cuda/nn/shrink.h]: Fixed.

File: onnxruntime/core/providers/cuda/nn/shrink_impl.cu [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    const T* input_data,
+    const float bias,
+    const float lambda,
+    T* output_data,
+    const CUDA_LONG N) {
+  CALCULATE_ELEMENTWISE_INDEX_OR_EXIT(id, N);
+
+  T x = input_data[id];
+  if (x < -lambda) {
+    output_data[id] = (T)(x + bias);
```
Comment (HectorSVC) [inline onnxruntime/core/providers/cuda/nn/shrink_impl.cu]: >bias [](start = 30, length = 4)

what about force cast bias to T? so no need the specialized implementation for half?

Comment (hariharans29) [reply onnxruntime/core/providers/cuda/nn/shrink_impl.cu]: Synced offline with Hector. This results in some unsafe casting and fails some unit tests. We need to rely on the compiler to do the implicit casting as the logic is more streamlined.

**Pull request merged**